Terminal.Gui/Core/View.cs

```diff
@@ -2372,32 +2467,38 @@ void CollectAll (View from, ref HashSet<View> nNodes, ref HashSet<(View, View)>
 			get => TextFormatter.Direction;
 			set {
 				if (TextFormatter.Direction != value) {
+					var isValidOldAutSize = autoSize && IsValidAutoSize (out Size autSize);
+					var directionChanged = TextFormatter.IsHorizontalDirection (TextFormatter.Direction)
+					    != TextFormatter.IsHorizontalDirection (value);
+
 					TextFormatter.Direction = value;
-					if (AutoSize) {
-						ResizeView (true);
-					} else if (IsInitialized) {
-						var b = new Rect (Bounds.X, Bounds.Y, Bounds.Height, Bounds.Width);
-						SetWidthHeight (b);
+					UpdateTextFormatterText ();
+
+					if ((!ForceValidatePosDim && directionChanged && AutoSize)
+					    || (ForceValidatePosDim && directionChanged && AutoSize && isValidOldAutSize)) {
+						ProcessResizeView ();
+					} else if (directionChanged && IsAdded) {
+						SetWidthHeight (Bounds.Size);
+						SetMinWidthHeight ();
+					} else {
+						SetMinWidthHeight ();
 					}
-					TextFormatter.Size = Bounds.Size;
+					TextFormatter.Size = GetBoundsTextFormatterSize ();
 					SetNeedsDisplay ();
 				}
 			}
 		}
 
-		bool isInitialized;
-
 		/// <summary>
 		/// Get or sets if  the <see cref="View"/> was already initialized.
 		/// This derived from <see cref="ISupportInitializeNotification"/> to allow notify all the views that are being initialized.
 		/// </summary>
-		public virtual bool IsInitialized {
-			get => isInitialized;
-			set {
-				isInitialized = value;
-				SetMinWidthHeight ();
-			}
-		}
+		public virtual bool IsInitialized { get; set; }
+
+		/// <summary>
+		/// Gets information if the view was already added to the superview.
+		/// </summary>
+		public bool IsAdded { get; private set; }
```

Comment:
Why `IsAdded` and not `IsAddedToSuperView`?

Reply:
Because in this logic would also have to change the following names:

`Add` to `AddToSuperView`
`Added` to `AddedToSuperView`
`OnAdded` to `OnAddedToSuperView`

But I changed the documentation to:
```cs
		/// <summary>
		/// Gets information if the view was already added to the <see cref="SuperView"/>.
		/// </summary>
```